core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -1804,6 +1798,21 @@ class WorkspaceService(protected val userInfo: UserInfo, val dataSource: SlickDa
     }
   }
 
+  def getGenomicsOperation(workspaceName: WorkspaceName, jobId: String): Future[PerRequestMessage] = {
+    // First check the workspace context and permissions in a DB transaction.
+    // We don't need any DB information beyond that, so just return Unit on success.
+    dataSource.inTransaction { dataAccess =>
+      withWorkspaceContextAndPermissions(workspaceName, WorkspaceAccessLevels.Read, dataAccess) { _ =>
+        DBIO.successful(())
+      }
+    }
+    // Next call GenomicsService, which actually makes the Google Genomics API call.
+    .flatMap { _ =>
+      val genomicsServiceRef = context.actorOf(GenomicsService.props(genomicsServiceConstructor, userInfo))
+      (genomicsServiceRef ? GenomicsService.GetOperation(jobId)).mapTo[PerRequestMessage]
```

Comment:
Yes! Always use `mapTo` instead of `asInstanceOf` to safely cast a Future to an expected type.
(It's often needed with Akka because actors are untyped 😢 )